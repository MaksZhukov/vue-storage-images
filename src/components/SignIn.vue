<template>
  <div class="signin">
    <div class="row">
      <form class="col s12" @submit.prevent="signIn({mail, pass})">
        <div class="row">
          <div class="col s12">
            <div class="input-field">
              <input id="email" type="email" class="validate" required v-model="mail">
              <label for="email">Email</label>
            </div>
            <div class="input-field">
              <input id="pass" type="password" class="validate" required v-model="pass">
              <label for="pass">Password</label>
            </div>
            <div class="row">
              <div class="input-field col">
                <button type="submit" class="btn pulse">Sign In</button>
              </div>
              <div class="input-field col">
                <button type="button" class="btn pulse red" @click="signInWithGoogle()">Sign In With Google</button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</template>

<script>
import firebase from "firebase";
import { mapActions, mapGetters } from "vuex";
import { toast } from "materialize-css";

export default {
  name: "SignIn",
  data: () => {
    return {
      mail: "",
      pass: ""
    };
  },
  methods: {
    ...mapActions("userModule", ["signIn", "signInWithGoogle"])
  },
  computed: mapGetters("userModule", { signInResponse: "getSignInResponse" })
};
</script>

<style lang="sass" scoped>
.signin
	max-width: 600px
	width: 100%
	margin: auto
</style>
