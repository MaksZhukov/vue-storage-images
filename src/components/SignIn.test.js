import { mount } from '@vue/test-utils'
import SignIn from './SignIn'

describe('SignIn component', () => {
  it('SignIn component render correctly', () => {
    const wrapper = mount(SignIn)
    expect(wrapper).toMatchSnapshot()
  })
  it('SignIn should not have props', () => {
    const wrapper = mount(SignIn)
    expect(Object.keys(wrapper.props()).length).toBe(0)
  })
  it('SignIn should contain button sign in', () => {
    const wrapper = mount(SignIn)
    expect(wrapper.html()).toContain('<button type="submit" class="btn pulse">Sign In</button>')
  })
  it('SignIn should have data pass and mail', () => {
    const wrapper = mount(SignIn)
    expect(wrapper.vm.mail).toBe('')
    expect(wrapper.vm.pass).toBe('')
  })
  it('SignIn should have button signInWithGoogle with text Sign In with Google', () => {
    const wrapper = mount(SignIn)
    const btnSignInWithGoogle = wrapper.find('.sign-in-with-google')
    expect(btnSignInWithGoogle.exists()).toBe(true)
    expect(btnSignInWithGoogle.text()).toBe('Sign In With Google')
  })
})
