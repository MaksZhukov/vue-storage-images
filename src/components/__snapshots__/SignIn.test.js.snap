// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component SignIn component render correctly 1`] = `
<div class="signin">
  <div class="row">
    <form class="col s12">
      <div class="row">
        <div class="col s12">
          <div class="input-field"><input id="email" type="email" required="required" class="validate"> <label for="email">Email</label></div>
          <div class="input-field"><input id="pass" type="password" required="required" class="validate"> <label for="pass">Password</label></div>
          <div class="row">
            <div class="input-field col"><button type="submit" class="btn pulse">Sign In</button></div>
            <div class="input-field col"><button type="button" class="btn pulse red sign-in-with-google">Sign In With Google</button></div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
